core/src/main/java/org/testcontainers/containers/ParsedDockerComposeFile.java

```diff
@@ -0,0 +1,103 @@
+package org.testcontainers.containers;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.EqualsAndHashCode;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Representation of a docker-compose file, with partial parsing for validation and extraction of a minimal set of
+ * data.
+ */
+@Slf4j
+@EqualsAndHashCode
+class ParsedDockerComposeFile {
+
+    private Map<String, Object> composeFileContent;
+    private final String composeFileName;
+    private Set<String> imageNames = new HashSet<>();
+
+    ParsedDockerComposeFile(File composeFile) {
+        Yaml yaml = new Yaml();
+        try (FileInputStream fileInputStream = FileUtils.openInputStream(composeFile)) {
+            composeFileContent = yaml.load(fileInputStream);
+        } catch (Exception e) {
+            throw new IllegalStateException("Unable to parse YAML file from " + composeFile.getAbsolutePath(), e);
+        }
+        this.composeFileName = composeFile.getAbsolutePath();
+
+        parseAndValidate();
+    }
+
+    @VisibleForTesting
+    ParsedDockerComposeFile(Map<String, Object> testContent) {
+        this.composeFileContent = testContent;
+        this.composeFileName = "";
+
+        parseAndValidate();
+    }
+
+    private void parseAndValidate() {
+        if (composeFileContent == null) {
```

Comment:
Is this necessary?
`yaml.load()` will return something if the yaml is valid and the other constructor is only available for testing.

Reply:
Good point - I think this was a leftover. Removing.